### PR TITLE
[INLONG-4636][CI] Add check PR title workflow

### LINKED ISSUE #4636: [Feature][CI] Add check PR title workflow
### Description

Add check PR title workflow.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shink**
Fixes #4636

### Motivation

Add check PR title workflow.

### Modifications

Add check PR title workflow.
The correct title should start with `[WIP]` or `[INLONG-xxx]`, where **xxx** is the issue number, e.g. `[INLONG-123]`.

### CODE REVIEW

File: .github/workflows/ci_check_pr_title.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      # where xxx is the issue number, e.g. [INLONG-123].
+      - name: Check title
+        run: |
+          if [[ ! "$PR_TITLE" =~ ^\[(WIP|INLONG-[0-9]+)\].*$ ]]; then
+            echo "This pull request title is not valid."
+            echo "The correct title should start with [WIP] or [INLONG-xxx], where xxx is the issue number, e.g. [INLONG-123]."
+            exit 1
+          fi
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
```
Comment (shink) [inline .github/workflows/ci_check_pr_title.yaml:46]: > The correct title should start with [WIP] or [INLONG-xxx], where xxx is the issue number, e.g. [INLONG-123].

Is it OK?

File: .github/workflows/ci_check_pr_title.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Check title
+        if: ${{ github.event_name == 'pull_request_target' }}
+        run: |
+          if [[ ! "$PR_TITLE" =~ ^\[(WIP|INLONG-[0-9]+)\].*$ ]]; then
+            echo "This pull request title is not valid."
+            echo "The correct title should start with [WIP] or [INLONG-xxx], where xxx is the issue number, e.g. [INLONG-123]."
+          
+            # add a job summary
+            echo "## This pull request title is not valid! :bug:" >> $GITHUB_STEP_SUMMARY
+            echo "The correct title should start with \`[WIP]\` or \`[INLONG-xxx]\`, where **xxx** is the issue number, e.g. \`[INLONG-123]\`." >> $GITHUB_STEP_SUMMARY
```
Comment (shink) [inline .github/workflows/ci_check_pr_title.yml:47]: If the title is not valid, the job summary will be added.
image

Comment (shink) [reply .github/workflows/ci_check_pr_title.yml:47]: See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

File: .github/workflows/ci_check_pr_title.yaml (hunk 1 above)
Comment (dockerzhang) [reply .github/workflows/ci_check_pr_title.yaml]: - Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/incubator-inlong/issues) number)*

File: .github/workflows/ci_check_pr_title.yml [hunk 3]
```diff
@@ -38,13 +38,17 @@ jobs:
...
+            echo "Title Example: [INLONG-XYZ][Component] Title of the pull request"
+            echo "XYZ should be replaced by the actual GitHub Issue number, e.g. [INLONG-123]"
+            echo "Component should be replaced by the InLong component name, e.g. [INLONG-123][Manager]"
           
             # add a job summary
             echo "## This pull request title is not valid! :bug:" >> $GITHUB_STEP_SUMMARY
-            echo "The correct title should start with \`[WIP]\` or \`[INLONG-xxx]\`, where **xxx** is the issue number, e.g. \`[INLONG-123]\`." >> $GITHUB_STEP_SUMMARY
+            echo "Title Example: \`[INLONG-XYZ][Component] Title of the pull request\`" >> $GITHUB_STEP_SUMMARY
+            echo "> - **XYZ** should be replaced by the actual [GitHub Issue](https://github.com/apache/incubator-inlong/issues) number, e.g. \`[INLONG-123]\`" >> $GITHUB_STEP_SUMMARY
+            echo "> - **Component** should be replaced by the InLong component name, e.g. \`[INLONG-123][Manager]\`" >> $GITHUB_STEP_SUMMARY
```
Comment (shink) [inline .github/workflows/ci_check_pr_title.yml:51]: @dockerzhang Done.

Comment (shink) [reply .github/workflows/ci_check_pr_title.yml:51]: image

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**